projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss

```diff
@@ -1233,23 +1228,86 @@
             height: rem(15px);
             min-width: rem(15px); /* yeah IE, it really needs to be 15px wide... */
             font-size: rem(15px);
-            color: igx-color($palette, 'grays', 500);
+        }
+    }
 
-            &:hover {
-                color: igx-color($palette, 'secondary', 500);
+    %igx-grid__th--sortable {
+        &:hover {
+            cursor: pointer;
+
+            %grid-cell-header-title {
+                cursor: pointer;
+                opacity: 1;
+            }
+
+            .sort-icon {
+                opacity: .7;
+
+                &:hover {
+                    opacity: 1;
+                }
             }
         }
     }
 
     %igx-grid__th--sorted {
-        color: igx-color($palette, 'grays', 900);
+        %grid-cell-header-title {
+            opacity: 1;
+        }
+
+        &:hover{
+            .sort-icon {
+                opacity: 1;
+            }
+        }
 
         .sort-icon {
             opacity: 1;
-            color: igx-color($palette, 'secondary', 500);
+            color: --var($theme, 'sorted-header-icon-color');
 
             &:hover {
-                color: igx-color($palette, 'secondary', 600);
+                color: --var($theme, 'sortable-header-icon-hover-color');
+            }
+        }
+    }
+
+    %igx-grid__th--filtrable {
+        %grid-cell-header-title {
+            opacity: .7;
+        }
+
+        &%igx-grid__th--sortable {
+            &:hover {
+                .sort-icon {
+                    &:hover {
+                        opacity: 1;
+                    }
+                }
+            }
+        }
+    }
+
+    %igx-grid__th--filtrable-open {
+        &%igx-grid__th--sortable,
+        &%igx-grid__th--sorted
+        {
+            %grid-cell-header-title {
+                color: --var($theme, 'filtering-header-text-color');
+                opacity: .7;
+            }
+
+            &:hover {
+                cursor: default;
+
+                %grid-cell-header-title {
+                    cursor: default;
```

Comment:
Same as the previous comment. Would be nice if we could limit the nesting to 2, max 3 levels.

Reply:
I am removing the styles for --filterable-open since we don't actually need them.